src/protocol/libp2p/kademlia/routing_table.rs

```diff
@@ -187,9 +187,11 @@ impl RoutingTable {
 
     /// Get `limit` closest peers to `target` from the k-buckets.
     pub fn closest<K: Clone>(&mut self, target: &Key<K>, limit: usize) -> Vec<KademliaPeer> {
-        ClosestBucketsIter::new(self.local_key.distance(target))
-            .flat_map(|index| self.buckets[index.get()].closest_iter(target))
+        ClosestBucketsIter::new(self.local_key.distance(&target))
+            .map(|index| self.buckets[index.get()].closest_iter(&target))
+            .flatten()
```

Comment:
Doesn't clippy complain about not using `flat_map()` here?

Reply:
i have changed it to flat_map now